### ISSUE TITLE
docker-credential-helpers: on linux fallback to `none` if `pass` isn't available

### ISSUE THREAD

**Issue by ericpromislow** [rancher-sandbox/rancher-desktop#2624]
### Actual Behavior

This should be done in the Diagnostics checker, not in the standard startup workflow

Not considering `secretservice`, because we use that only if it's specified. This is done when we're looking
to see whether to default to `pass` or `none`.

Note that `docker-credential-none` stores creds in unencrypted base64 -- the user needs to be warned about that.

### Steps to Reproduce

f

### Result

f

### Expected Behavior

f

### Additional Information

f

### Rancher Desktop Version

f

### Rancher Desktop K8s Version

f

### Which container runtime are you using?

containerd (nerdctl)

### What operating system are you using?

Other Linux

### Operating System / Build Version

fds

### What CPU architecture are you using?

x64

**Comment by jandubois** [comment]
> This should be done in the Diagnostics checker, not in the standard startup workflow

No, in the standard setup workflow we will test if `pass` can work (it may be as simple as running `docker-credential-pass list` and checking if it returns valid JSON).

If `pass` doesn't work, then we will configure `none` automatically.

The diagnostics check will notice that `none` has been configured and display a warning that credentials are stored in plaintext.
